Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -169,6 +174,11 @@ class TopSitesPanel: UIViewController {
         // now update the data source with the new data
         self.updateDataSourceWithSites(result)
 
+        // Exit early if the query failed in some way.
+        guard result.isSuccess else {
```

Comment:
You can actually move this before the `self.updateDataSourceWithSites` call, too.